packages/devtools_app/lib/src/timeline/timeline_model.dart

```diff
@@ -678,11 +675,48 @@ abstract class TimelineEvent extends TreeNode<TimelineEvent> {
   /// time of any of its descendant events.
   int get maxEndMicros;
 
-  int get displayDepth;
+  bool couldBeParentOf(TimelineEvent e);
 
-  bool get hasOverlappingChildren;
+  /// Tracks the start row for the lowest visual child in the display for this
+  /// TimelineEvent.
+  int _lowestDisplayChildRow = 1;
 
-  bool couldBeParentOf(TimelineEvent e);
+  /// The lowest visual child in the display for this TimelineEvent.
```

Comment:
nit: no-op comment as written. Maybe tweak to explain what lowest means rather than using the same word as in the property name.